release.sh

```diff
@@ -186,7 +188,13 @@ if [ "$PUSH_CHANGES" != "true" ]; then
 	ADDITIONAL_OPTIONS="-d"
 fi
 
-bash -xe "$SCRIPTS_DIR/prepare-release.sh" "$PROJECT" "$RELEASE_VERSION"
+elif [ "$PROJECT" == "infra-theme" || "$PROJECT" == "infra-extensions" ]; then
+  # infra projects are using different release plugins then main maven-based projects,
+  # and they require both release and development versions:
+  bash -xe "$SCRIPTS_DIR/prepare-release.sh" "$PROJECT" "$RELEASE_VERSION" "$DEVELOPMENT_VERSION"
+else
+  bash -xe "$SCRIPTS_DIR/prepare-release.sh" "$PROJECT" "$RELEASE_VERSION"
```

Comment:
why don't you always pass the extra parameter?
If it's not needed the other scripts will ignore it.

Reply:
we had a trouble with that 3d parameter in the past 😃 
see here:
https://github.com/hibernate/hibernate-release-scripts/blob/41bf900e72653d699d970457c89aabbf9d148465/prepare-release.sh#L7

and then if that inherited version is later passed here:

https://github.com/hibernate/hibernate-release-scripts/blob/41bf900e72653d699d970457c89aabbf9d148465/prepare-release.sh#L63

then the wrong version update is called and all fails 🙈 😃 

So that's why I thought I'll make it like that ...

Reply:
I would always set 4 parameters and then validate that all of them are passed.
Not sure if empty strings are a problem though.

Otherwise you'll end up with the same problem if somebody is calling `prepare-release.sh` from some other file and it's not aware of these issues.

But I'm not gonna fight, if you are happy with this solution.

Reply:
Oh, I'd be happy to make it cleaner :smile: I'm just not sure if I change the inputs that I'm not going to break some other project setup ...
I'm not even sure where this inherited version comes from 🙈 
I'll take another look at this to see if I can make it cleaner and thanks for suggestions

Reply:
Is this, latest update, what you had in mind? 🙈 😃

Reply:
What do you mean?

Reply:
> What do you mean?

🙈 I meant that I tried to apply your suggestion 😃 and pushed the updates to this PR 😃 

> Anyway, I think we call prepare-release in the Jenkins pipeline configuration too, would that cause any issue?
See https://github.com/hibernate/hibernate-orm/blob/7e4df3f805c6155ba3f534230958862b1fbbdba6/ci/release/Jenkinsfile#L182

Yeah, if we apply the current change, we'd need to adjust the ORM release Jenkins files in all branches with an "automated release" setup... 🙈

Reply:
> Maybe your initial solution is the best one at the moment.

mm ok... so maybe, let me revert back to that for now, and I'll try to do the infra releases and we can come back to make the scripts cleaner at a later point ...